test/famfs_unit.c

```diff
@@ -0,0 +1,113 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2023-2024 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <unistd.h>
+#include <linux/limits.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <linux/famfs_ioctl.h>
+#include "famfs_lib.h"
+#include "famfs_lib_internal.h"
+#include "famfs_meta.h"
+#include "xrand.h"
+#include "random_buffer.h"
+#include "famfs_unit.h"
+
+//typedef uint64_t u64;
+
+
+#define famfs_assert_eq(a, b) {						\
+	if ((a) != (b))	{						\
+		fprintf(stderr, "%s:%d %lld != %lld\n",			\
+		   __FILE__, __LINE__, (u64)(a), (u64)(b));		\
+		return __LINE__;					\
+	}								\
+}
+
+#define famfs_assert_gt(a, b) {						\
+	 if ((a) <= (b))	{					\
+		 fprintf(stderr, "%s:%d %lld !<= %lld\n",		\
+			 __FILE__, __LINE__, (u64)(a), (u64)(b));	\
+		 return __LINE__;					\
+	 }								\
+}
+
+#define famfs_assert_ne(a, b) {						\
+	if ((a) == (b)) {						\
+		fprintf(stderr, "%s:%d %lld == %lld\n",			\
+			__FILE__, __LINE__, (u64)(a), (u64)(b));	\
+		return __LINE__;					\
+	}								\
+}
+
+int create_mock_famfs_instance(
```

Comment:
This can be called repeatedly by unit tests to create a mocked famfs instance that has a working superblock and log